src/components/QuestionListData.tsx

```diff
@@ -0,0 +1,130 @@
+import { LoadingOutlined } from '@ant-design/icons';
+import { Input, Rate, Space, Spin } from 'antd';
+import { ColumnsType } from 'antd/lib/table';
+import { FilterConfirmProps } from 'antd/lib/table/interface';
+import React, { useRef } from 'react';
+
+import { InterviewQuestionModel, QuestionListModel } from '../services/Client';
+import {
+  createColumnFilterProps,
+  createColumnSearchProps,
+} from '../utils/tableUtils';
+import styles from './QuestionLists.module.scss';
+import TableWrapper from './TableWrapper';
+
+export default function QuestionListData({
+  isBeingEdited,
+  list,
+  displayableQuestions,
+  addableQuestions,
+  allQuestions,
+  addToRemoveDrawerCallback,
+  addToAddDrawerCallback,
+}: {
+  isBeingEdited: boolean;
+  list: QuestionListModel | undefined;
+  displayableQuestions: InterviewQuestionModel[];
+  addableQuestions: InterviewQuestionModel[];
+  allQuestions: InterviewQuestionModel[];
+  addToRemoveDrawerCallback: (question: InterviewQuestionModel) => void;
+  addToAddDrawerCallback: (question: InterviewQuestionModel) => void;
+}): JSX.Element {
+  const searchInput = useRef<Input>();
+
+  const handleSearch = (
+    confirm: (param: FilterConfirmProps | undefined) => void
+  ): void => {
+    confirm({ closeDropdown: true });
+  };
+
+  const handleReset = (
+    clearFilters: () => void,
+    confirm: (param: FilterConfirmProps | undefined) => void
+  ): void => {
+    clearFilters();
+    confirm({ closeDropdown: true });
+  };
+
+  const tableColumns = (
+    data: InterviewQuestionModel[]
+  ): ColumnsType<InterviewQuestionModel> => {
+    return [
+      {
+        ...createColumnSearchProps(
+          'title',
+          searchInput,
+          handleSearch,
+          handleReset
+        ),
+      },
+      {
+        ...createColumnFilterProps(
+          'category',
+          data
+            .map((value) => value.category!.toString())
+            .filter((value) => value !== undefined)!
+        ),
+      },
+      {
+        ...createColumnSearchProps(
+          'content',
+          searchInput,
+          handleSearch,
+          handleReset
+        ),
+      },
+      {
+        ...createColumnFilterProps(
+          'difficulty',
+          data
+            .map((value) => value.difficulty!.toString())
+            .filter((value) => value !== undefined)!
+        ),
+        sorter: (a: InterviewQuestionModel, b: InterviewQuestionModel) =>
+          a.difficulty != null && b.difficulty != null
```

Comment:
It is not necessary to compare an object with null value, you can write this condition easier.
`a.difficulty && b.difficulty`